packages/synapse-react-client/src/components/ColumnFilter/ColumnFilter.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+import { useQueryContext } from '../QueryContext/QueryContext'
+import {
+  FacetColumnResultValues,
+  FacetColumnValuesRequest,
+} from '@sage-bionetworks/synapse-types'
+import { useQuery } from '@tanstack/react-query'
+import { Autocomplete, TextField } from '@mui/material'
+import {
+  facetObjectMatchesDefinition,
+  getCorrespondingSelectedFacet,
+} from '../../utils/functions/queryUtils'
+import { UniqueFacetIdentifier } from '../../utils'
+
+export type FilterProps = {
+  topLevelEnumeratedFacetToFilter: UniqueFacetIdentifier
+}
+
+function ColumnFilter(props: FilterProps) {
+  const queryContext = useQueryContext()
+  const {
+    executeQueryRequest,
+    queryMetadataQueryOptions,
+    getCurrentQueryRequest,
+    addValueToSelectedFacet,
+    removeSelectedFacet,
+  } = queryContext
+  const { data: queryMetadata } = useQuery(queryMetadataQueryOptions)
+  const { topLevelEnumeratedFacetToFilter } = props
+
+  const currentQuery = getCurrentQueryRequest()
+
+  const activeFacet = (queryMetadata?.facets ?? []).find(facet =>
+    facetObjectMatchesDefinition(facet, topLevelEnumeratedFacetToFilter),
+  ) as FacetColumnResultValues | undefined
+
+  // Map the facet values to format for MUI Autocomplete options
+  const filterOptions = activeFacet?.facetValues.map(({ value }) => value) ?? []
+
+  const selectedFacetFromQuery = getCorrespondingSelectedFacet(
+    { columnName: topLevelEnumeratedFacetToFilter.columnName },
```

Comment:
You can just pass this since it is a facet identifier
```suggestion
    topLevelEnumeratedFacetToFilter,
```